src/plugins/event-plugins/DomEventPlugin.ts

```diff
@@ -27,16 +27,19 @@ export type TargetDomEvent = {
 };
 
 export type PartialDomEventPluginConfig = {
+    customIdTracker?: (event: Event) => string;
     enableMutationObserver?: boolean;
     events?: TargetDomEvent[];
 };
 
 export type DomEventPluginConfig = {
+    customIdTracker?: (event: Event) => string;
     enableMutationObserver?: boolean;
     events: TargetDomEvent[];
 };
 
 const defaultConfig: DomEventPluginConfig = {
+    customIdTracker: undefined,
```

Comment:
suggestion: For all config options, we define sane default. Define a default function here and have it return `undefined`. Then we can get rid of the `if` statement at line 140.
```suggestion
    customIdTracker: () => undefined,
```

Reply:
Sounds like a better approach. Will add this.